app/src/main/java/tech/relaycorp/courier/domain/client/CargoCollection.kt

```diff
@@ -35,18 +35,21 @@ class CargoCollection
     private suspend fun getCCAs() =
         storedMessageDao.getByRecipientTypeAndMessageType(
             MessageAddress.Type.Public,
-            MessageType.Cargo
+            MessageType.CCA
         )
 
     private suspend fun collectAndStoreCargoForCCA(cca: StoredMessage) {
+        val client = clientBuilder.build(cca.recipientAddress.value)
         try {
-            val client = clientBuilder.build(cca.recipientAddress.value)
             client
                 .collectCargo(cca.getSerializedInputStream())
                 .collect { storeCargo(it) }
-            client.close()
         } catch (e: MessageDataNotFoundException) {
             logger.log(Level.WARNING, "CCA data could not found on disk", e)
+        } catch (e: CogRPCClient.CCARefusedError) {
+            logger.log(Level.WARNING, "CCA refused")
+        } finally {
+            client.close()
```

Comment:
We're not catching other CogRPC-related exceptions here. What effect would that have on the UX and the overall sync with all public gateways?

I'm just trying to check if we need to catch all other CogRPC exceptions to ignore them after logging them.

Reply:
The `PublicSync` is handling all remaining scenarios, and switching the UI to an Error state, which we haven't designed yet.

Reply:
As it is, would it cause syncs with other public gateways to be skipped or interrupted? Irrespective of how we handle such errors in the UI, a failure with one gateway shouldn't affect syncs with other gateways.

I've just added a TODO to #16 to make sure we'll handle this in the UI, but I think we need to make sure that a failed sync won't disrupt other syncs in this PR. I'm not saying this function is the right place to do that, but I think we need to handle it somewhere in this PR.